src/attr.rs

```diff
@@ -24,12 +29,59 @@ pub fn valid<I: Iterator<Item = char>>(chars: I) -> (usize, bool) {
     (p.pos, has_attr)
 }
 
+pub struct AttrChars<'s> {
+    chars: Peekable<Chars<'s>>,
+}
+
+impl<'s> Iterator for AttrChars<'s> {
+    type Item = char;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some(c) = self.chars.next() {
+            match c {
+                '\\' => {
+                    if let Some('\\') = self.chars.peek() {
+                        return Some('\\');
+                    }
+                }
+                _ => return Some(c),
```

Comment:
do we want to match the reference implementation by only  escaping ascii punctuation? E.g.

- `\a` -> `\a`
- `\.` -> `.`
- `\\` -> `\`

The lexer behaves like this also, so it might be more "unified" to match this behavior.

Reply:
Done!